components/usage/pkg/scheduler/job.go

```diff
@@ -28,40 +28,8 @@ func NewPeriodicJobSpec(period time.Duration, id string, job Job) (JobSpec, erro
 	}
 
 	return JobSpec{
-		Job:      WithoutConcurrentRun(job),
+		Job:      job,
 		ID:       id,
 		Schedule: parsed,
 	}, nil
 }
-
-// WithoutConcurrentRun wraps a Job and ensures the job does not concurrently
-func WithoutConcurrentRun(j Job) Job {
```

Comment:
I liked how you made non concurrent execution a feature of the Job API [here](https://github.com/gitpod-io/gitpod/pull/17433/files#diff-2d1814d4e79d79146e0a3be29f6118d6e93de6bfb60c8f4ce44f93c5a3ccdfbd). Wouldn't it make sense to do it similarly for our go components?

Reply:
Sure, I can do that. Originally, the idea was to keep them separate and composable but it looks like in practice we don't actually have a use-case for concurrent runs.

I can refactor it in a follow-up PR to provide a common interface. Landing this as is unblocks running multiple replicas of usage and eventually merging it with usage so would like to land this as is to unblock those